addons/hr_expense/models/product_product.py

```diff
@@ -24,3 +24,33 @@ def _compute_standard_price_update_warning(self):
                             "There are unsubmitted expenses linked to this category. Updating the category cost will change expense amounts. "
                             "Make sure it is what you want to do."
                         )
+
+    def write(self, vals):
+        res = super().write(vals)
+        if 'standard_price' not in vals:  # Only check for expenses if the standard price has been updated
+            return res
+
+        expenses_sudo = self.env['hr.expense'].sudo().search([
+            ('state', 'in', ['draft', 'reported']),
+            ('product_id', 'in', self.ids),
+        ])
+        for expense_sudo in expenses_sudo:
+            expense_product = expense_sudo.product_id.sudo(False)
+            tax_domain = self.env['account.tax']._check_company_domain(expense_sudo.company_id)
```

Comment:
What's the added value of calling this function specifically instead of just building the domain ? Plus, it does not seem to consider branches, does it ? :thinking:

Reply:
Indeed, fixed in Ali's PR and rebased